src/Compiler/Microsoft.AspNetCore.Razor.Language/test/MetadataCollectionTests.cs

```diff
@@ -35,7 +35,9 @@ public void CreateAndCompareCollections(int size)
         }
 
         var collection1 = MetadataCollection.Create(pairs.ToArray());
-        var collection2 = MetadataCollection.Create(pairs.ToArray().Reverse().ToArray());
+        // force conversion to IEnumerable so Linq Reverse method gets called instead of the new System.MemoryExtensions.Reverse
+        var enumerableCollection = (IEnumerable<KeyValuePair<string, string?>>)pairs;
+        var collection2 = MetadataCollection.Create(enumerableCollection.Reverse().ToArray());
```

Comment:
@ryzngard has a more straightforward fix for this issue in their PR [here](https://github.com/dotnet/razor/pull/11096/files#diff-346d9426964986b379c17db92da94a1ef35d54f40a92a852f37238773e151e6f). I recommend taking that change:

```suggestion

        var reversed = Enumerable.Reverse(pairs.ToArray());
        var collection2 = MetadataCollection.Create(reversed.ToArray());
```

Reply:
Modified to match